storage/innobase/include/sess0sess.h

```diff
@@ -177,9 +209,12 @@ class innodb_session_t {
 
   /** Current session's user temp tablespace */
   ibt::Tablespace *m_usr_temp_tblsp;
-
+  /** Current session's encrypted user temp tablespace */
+  ibt::Tablespace *m_enc_usr_temp_tblsp;
```

Comment:
Why we need to have both m_usr_temp_tblsp and m_enc_usr_temp_tblsp, likewise for intrinsic? Can user have encrypted and unencrypted temp tables simultaneously?

Reply:
yes, it is possible
only on SET GLOBAL innodb_encrypt_tables=ON or SET GOBAL innodb_temp_tablespace_encrypt=ON the tables will go the encrypted session temp tablespaces
until then, all tables would have gone to unencrypted session temp tablespaces.